code/modules/mob/living/simple_animal/hostile/pulse_demon/pulsedemon.dm

```diff
@@ -78,16 +79,36 @@
 		forceMove(current_power)
 	set_light(1.5,2,"#bbbb00")
 	add_spell(new /spell/pulse_demon/abilities, "pulsedemon_spell_ready", /obj/abstract/screen/movable/spell_master/pulse_demon)
-	add_spell(new /spell/pulse_demon/toggle_drain, "pulsedemon_spell_ready", /obj/abstract/screen/movable/spell_master/pulse_demon)
+	var/datum/action/pd_toggle_drain/PTD = new(src)
+	PTD.Grant(src)
+	PLI = new(src)
 	for(var/pd_spell in getAllPulseDemonSpells())
 		var/spell/S = new pd_spell
-		if(S.type != /spell/pulse_demon && S.type != /spell/pulse_demon/abilities && S.type != /spell/pulse_demon/toggle_drain)
+		if(S.type != /spell/pulse_demon && S.type != /spell/pulse_demon/abilities)
 			possible_spells += S
 	for(var/pd_upgrade in subtypesof(/datum/pulse_demon_upgrade))
 		var/datum/pulse_demon_upgrade/PDU = new pd_upgrade(src)
 		possible_upgrades += PDU
 	playsound(get_turf(src),'sound/effects/eleczap.ogg',50,1)
 
+/mob/living/simple_animal/hostile/pulse_demon/maxedout // For testing it, maybe other fun reasons too
+	charge = INFINITY
+	maxcharge = INFINITY
+	health_drain_rate = 1
+	health_regen_rate = INFINITY
+	amount_per_regen = 1
+	charge_absorb_amount = INFINITY
+//	max_can_absorb = INFINITY
+	takeover_time = 1
+	move_divide = 1
+
+/mob/living/simple_animal/hostile/pulse_demon/maxedout/New()
+	..()
+	for(var/spell/S in possible_spells)
```

Comment:
you can use pick_n_take here

Reply:
howso?

Reply:
```var/spell/S = pick_n_take(possible_spells)
add_spell(S,otherargs)```

Reply:
won't that add them in a random order though?

Reply:
You pose a good point, if you care about the order you really shouldn't use that helper here which is explicitly trying to randomize

Reply:
i'd rather not mess with a tester's muscle memory so order is a nicer thing to have